src/util/CartesianUtils.ts

```diff
@@ -238,3 +238,18 @@ export const createLabeledScales = (options: Record<string, any>): LabeledScales
     },
   } as LabeledScales<Record<string, any>>;
 };
+
+/** Calculates the width of the largest horizontal line that fits inside a rectangle that is displayed at an angle.
+ * @param {Object} size Width and height of the text in a horizontal position.
+ * @param {number} angle Angle in degrees in which the text is displayed.
+ * @return {number} The width of the largest horizontal line that fits inside a rectangle that is displayed at an angle.
+ */
+export const getAngledRectangleWidth = ({ width, height }: Size, angle: number | undefined = 0) => {
```

Comment:
This calculates the width of the red arrow shown below:

![image](https://github.com/recharts/recharts/assets/12778398/2469f81c-2b5c-4ada-8541-36ce37d3a059)

This width is the effective width of the tick and is what will be used to decide whether to show a tick.

Reply:
Could you please a a few unit tests to this function, to document the behaviour and test edge cases?